capellambse/loader/core.py

```diff
@@ -296,10 +325,30 @@ def __init__(
             pathlib.PurePosixPath("\0", self.entrypoint)
         )
 
+        self.check_duplicate_uuids()
+
     @property
     def filehandler(self) -> filehandler.FileHandler:
         return self.resources["\0"]
 
+    def check_duplicate_uuids(self):
+        seen_ids = set[str]()
```

Comment:
"The usual way" would actually be `seen_ids: t.Set[str] = set()` – what you wrote is already a shorthand possible since 3.8 with `__future__.annotations`, and since 3.9 it's also possible this way to avoid duplicating the type name at all (in some cases). Most notably this doesn't work with PEP-604-style unions on pre-3.10.